src/algebra/big_operators/order.lean

```diff
@@ -308,3 +309,70 @@ open opposite
 (@unop_add_hom β _).map_sum _ _
 
 end with_top
+
+namespace finset
+
+/--
+The strong pigeonhole principle for finitely many pigeons and
+pigeonholes: there is a pigeonhole with at least as many pigeons as
+the ceiling of the average number of pigeons across all pigeonholes.
+("The maximum is at least the mean" specialized to integers.)
+
+More formally, given a function between finite sets `s` and `t` and
+setting `n` so that `n + 1` is at most the ceiling of `s.card / t.card`,
+then there is an element of `t` whose preimage contains more
+than `n` elements.  (We formulate the constraint on `n` as
+`t.card * n < s.card`, which incorporates that `s` and `t` must be nonempty.)
```

Comment:
This inequality implies only `s.nonempty`. You need `hf` to deduce `t.nonempty`.

Reply:
I meant that in the context of there being a function between finite sets, but I can clarify this if you want.